src/wallet/wallet.cpp

```diff
@@ -2344,47 +2346,29 @@ bool CWalletTx::IsEquivalentTo(const CWalletTx& _tx) const
         return CTransaction(tx1) == CTransaction(tx2);
 }
 
-// Rebroadcast transactions from the wallet. We do this on a random timer
-// to slightly obfuscate which transactions come from our wallet.
-//
-// Ideally, we'd only resend transactions that we think should have been
-// mined in the most recent block. Any transaction that wasn't in the top
-// blockweight of transactions in the mempool shouldn't have been mined,
-// and so is probably just sitting in the mempool waiting to be confirmed.
-// Rebroadcasting does nothing to speed up confirmation and only damages
-// privacy.
+// Once a day, resbumit all wallet transactions to the node,
+// incase it has been dropped from your mempool.
 void CWallet::ResendWalletTransactions()
 {
     // During reindex, importing and IBD, old wallet transactions become
-    // unconfirmed. Don't resend them as that would spam other nodes.
+    // unconfirmed. Don't need to resubmit to our node.
     if (!chain().isReadyToBroadcast()) return;
 
-    // Do this infrequently and randomly to avoid giving away
-    // that these are our transactions.
-    if (GetTime() < nNextResend || !fBroadcastTransactions) return;
-    bool fFirst = (nNextResend == 0);
-    nNextResend = GetTime() + GetRand(30 * 60);
-    if (fFirst) return;
-
-    // Only do it if there's been a new block since last time
-    if (m_best_block_time < nLastResend) return;
-    nLastResend = GetTime();
+    // Do this once per day.
+    if (GetTime() < nNextResend) return;
+    nNextResend = GetTime() + RESEND_TXS_FREQUENCY;
 
     int submitted_tx_count = 0;
 
     { // locked_chain and cs_wallet scope
         auto locked_chain = chain().lock();
         LOCK(cs_wallet);
 
-        // Relay transactions
+        // Resubmit transactions
         for (std::pair<const uint256, CWalletTx>& item : mapWallet) {
             CWalletTx& wtx = item.second;
-            // Attempt to rebroadcast all txes more than 5 minutes older than
-            // the last block. SubmitMemoryPoolAndRelay() will not rebroadcast
-            // any confirmed or conflicting txs.
-            if (wtx.nTimeReceived > m_best_block_time - 5 * 60) continue;
             std::string unused_err_string;
-            if (wtx.SubmitMemoryPoolAndRelay(unused_err_string, true, *locked_chain)) ++submitted_tx_count;
+            if (wtx.SubmitMemoryPoolAndRelay(unused_err_string, false, *locked_chain)) ++submitted_tx_count;
```

Comment:
This looks good. There's some tidyup you could do afterwards (in a separate commit so it doesn't distract from the purpose of this commit):

- `SubmitMemoryPoolAndRelay()` is now only called with `relay=false`, so the function could be simplified (and potentially renamed since it doesn't actually relay now. The `Chain::broadcastTransaction()` interface method can also be simplified and renamed.
- `m_best_block_time` is now no longer used and can be removed. That means that the wallet no longer needs to subscribe to the `UpdatedBlockTip()` Validation Interface method!

Reply:
we discussed this, but for reference:
- `CommitTransaction` calls `SubmitMemoryPoolAndRelay` with `relay=true` 
- still theoretically could remove the boolean because the checks in `BroadcastTransaction` could decide whether or not to relay, but with my changes I need to force relay=false because the mempool would not be able to distinguish a resubmit from a new txn, and we would not immediately want to relay a resubmit because that would be a privacy leak

however, `m_best_block_time` can totally be removed, noted for a follow up.

also, this surfaced a bug with resubmitted txns being added to the unbroadcast set, thus failing to achieve that intended privacy. fixed on the bitcoin/bitcoin PR.